back/src/test/java/com/ubikloadpack/jmeter/ulp/observability/server/ULPObservabilityConfigServletTest.java

```diff
@@ -0,0 +1,51 @@
+package com.ubikloadpack.jmeter.ulp.observability.server;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.eclipse.jetty.http.HttpStatus;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import com.ubikloadpack.jmeter.ulp.observability.util.Util;
+
+public class ULPObservabilityConfigServletTest extends AbstractConfigTest {
+	@Test
+	@DisplayName("When a GET request is sent, expect ok and json as content type")
+	public void whenGetRequestExpectOkAndJsonContentType() throws Exception {	
+		HttpResponse httpResponse = this.sendGetRequest("/config");
+		assertHttpContentTypeAndResponseStatus(httpResponse, HttpStatus.OK_200, "application/json");
+	}
+	
+	@Test
+	@DisplayName("When sending GET request, expect the json format and the values of the received configuration")
+	public void whenSendingGetRequestToConfigEndpointExpectConfigurationAsJsonFormat() throws Exception {		
+		HttpResponse httpResponse = this.sendGetRequest("/config");
+		assertHttpContentTypeAndResponseStatus(httpResponse, HttpStatus.OK_200, "application/json");
+        
+        String actualConfig = httpResponse.getResponse();
+        String expectedConfig = getExpectedConfigAsJson(METRICS_ROUTE, LOG_FREQUENCY, TOP_ERRORS, TOTAL_LABEL);
+        
+        assertEquals(expectedConfig, actualConfig);
+	}
+	
+	@Test
+	@DisplayName("when totalLabel contains spaces expect formatted totalLabel in openMetrics format")
+	public void whenTotalLabelContainsSpacesExpectFormattedTotalLabelInOpenMetricsFormat() throws Exception {		
+		this.listener.testEnded(HOST);

```

Comment:
The sequence of events looks strange.
Why testStarted is called after testEnded ?

Reply:
Because there was a @BeforeEach that invokes the testStarted() and I was forced to stop the litsener (method testEnded()) to set the totalLabel property. Then, restart the test with testStarted().

Reply:
Can you add a comment to make this explicit ?

Reply:
ok